src/dune_engine/build_system.ml

```diff
@@ -1164,7 +1160,8 @@ end = struct
             (Memo.Stack_frame.as_instance_of
                frame
                ~of_:execute_action_generic_stage2_memo)
-            ~f:(fun (x : Anonymous_action.t) -> x.action.loc)))
+            ~f:(fun (x : Anonymous_action.t) ->
+              Option.some_if (not @@ Loc.is_none x.action.loc) x.action.loc)))
```

Comment:
This part is a little confusing. So we are turning a Loc back into a Loc option here? Why do we need to have an option type at all?